tests/test_energy_network.py

```diff
@@ -0,0 +1,128 @@
+        optimization_type = "costs"  # set as "env" for environmental optimization and "costs" for cost optimization
+
+        # When
+        _os.chdir(_examples_dir)
+        network = _en.EnergyNetworkClass(time_period)
+        network.setFromExcel(str(_input_data_path), nr_of_buildings, opt=optimization_type)
+        _os.chdir(cwd)
+
+        # Then
+        errors = []
+        check_assertion(self, errors, len(network.groups), 147)
+
+        check_assertion(self, errors, len(network.nodes), 140)
+
+        check_assertion(self, errors, network.results, None)
+
+        check_assertion(self, errors, len(network.signals), 1)
+
+        check_assertion(self, errors, network.temporal, None)
+
+        check_assertion(self, errors, len(network.timeincrement), 744)
+
+        check_assertion(self, errors, network.timeincrement.unique(), 1.0)
+
+        check_assertion(self, errors, len(network.timeindex), 745)
+
+        if errors:
+            raise ExceptionGroup(f"found {len(errors)} errors", errors)
+
+    def test_get_nodal_data_from_Excel(self):
+        """File System touching test to get an initial grip on what the nodal data should be after reading the
+        scenario."""
+        network = _en.EnergyNetworkClass(None)
+
+        data = _pd.ExcelFile(str(_input_data_path))
+        initial_nodal_data = network.get_nodal_data_from_Excel(data)
+        data.close()
+
+        errors = []
+
+        check_assertion(self, errors, len(initial_nodal_data), 9)
+        check_assertion(
+            self,
+            errors,
+            list(initial_nodal_data.keys()),
+            [
+                "buses",
+                "grid_connection",
+                "commodity_sources",
+                "solar",
+                "transformers",
+                "demand",
+                "storages",
+                "stratified_storage",
+                "profiles",
+            ],
+        )
+
+        check_assertion(self, errors, initial_nodal_data["buses"].shape, (48, 5))
+        check_assertion(self, errors, initial_nodal_data["grid_connection"].shape, (24, 5))
+        check_assertion(self, errors, initial_nodal_data["commodity_sources"].shape, (8, 6))
+        check_assertion(self, errors, initial_nodal_data["solar"].shape, (8, 31))
+        check_assertion(self, errors, initial_nodal_data["transformers"].shape, (16, 19))
+        check_assertion(self, errors, initial_nodal_data["demand"].shape, (12, 7))
+        check_assertion(self, errors, initial_nodal_data["storages"].shape, (12, 20))
+        check_assertion(self, errors, initial_nodal_data["stratified_storage"].shape, (2, 11))
+        check_assertion(self, errors, initial_nodal_data["profiles"].shape, (2, 3))
+
+        if errors:
+            raise ExceptionGroup(f"found {len(errors)} errors", errors)
+
+    def test_set_from_csvs(self):
+        """File system touching test.
+        Compare network state of the csv variety with that of the Excel variety.
+        """
```

Comment:
This couples the tests between xls and csv, which is not optimal.
It helped add set_from_csv however.